modules/programs/xplr.nix

```diff
@@ -1,30 +1,47 @@
 { config, lib, pkgs, ... }:
 let
   inherit (lib)
-    concatStringsSep types mkIf mkOption mkEnableOption mkPackageOption
-    literalExpression;
+    types mkIf mkOption mkEnableOption mkPackageOption literalExpression;
 
   cfg = config.programs.xplr;
 
   initialConfig = ''
     version = '${cfg.package.version}'
   '';
 
-  # We provide a default version line within the configuration file, which is
-  # obtained from the package's attributes. Merge the initial configFile, a
-  # mapped list of plugins and then the user defined configuration to obtain the
-  # final configuration.
-  pluginPath = if cfg.plugins != [ ] then
-    (''
-      package.path=
-    '' + (concatStringsSep " ..\n"
-      (map (p: ''"${p}/init.lua;${p}/?.lua;"'') cfg.plugins)) + ''
-         ..
-        package.path
-      '')
+  # If `value` is a Nix store path, create the symlink `/nix/store/newhash/${name}/*` 
+  # to `/nix/store/oldhash/*` and returns `/nix/store/newhash`.
+  wrapPlugin = name: value:
+    if lib.isStorePath value then
+      pkgs.symlinkJoin {
+        name = name;
+        paths = [ value ];
+        postBuild = ''
+          mkdir ${name}
+          mv $out/* ${name}/
+          mv ${name} $out/
```

Comment:
```suggestion
          mkdir './${name}'
          mv $out/* './${name}/'
          mv './${name}' $out/
```

Reply:
is mkdir even necessary here?

Reply:
We need to make a dir named "${name}" so that we can move the contents in there.

Reply:
Thank you. Resolved by latest force push.